frontend/src/components/SaveSearchModal/SaveSearchModal.tsx

```diff
@@ -0,0 +1,254 @@
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useSavedSearchContext } from 'context/SavedSearchContext';
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  TextField,
+  Button,
+  Box
+} from '@mui/material/';
+import { SavedSearch } from '../../types/saved-search';
+import { useAuthContext } from '../../context';
+import { Vulnerability } from 'types';
+import { Save } from '@mui/icons-material';
+
+interface SaveSearchModalProps {
+  search: any;
+  searchTerm: string;
+  filters: any;
+  totalResults: number;
+  sortField: string;
+  sortDirection: string;
+  advancedFiltersReq?: boolean;
+}
+
+export const SaveSearchModal: React.FC<SaveSearchModalProps> = (props) => {
+  const {
+    search,
+    searchTerm,
+    filters,
+    totalResults,
+    sortField,
+    sortDirection,
+    advancedFiltersReq
+  } = props;
+  const [open, setOpen] = useState(false);
+  const [dialogeOpen, setDialogOpen] = useState(false);
+  const [formErrors, setFormErrors] = useState({
+    name: false,
+    duplicate: false
+  });
+  const { apiPost, apiPut } = useAuthContext();
+  const history = useHistory();
+  const { savedSearches } = useSavedSearchContext();
+  const [savedSearchValues, setSavedSearchValues] = useState<
+    Partial<SavedSearch> & {
+      name: string;
+      vulnerabilityTemplate: Partial<Vulnerability>;
+    }
+  >(
+    search
+      ? search
+      : {
+          name: '',
+          vulnerabilityTemplate: {},
+          createVulnerabilities: false
+        }
+  );
+
+  // API call to save/update saved searches
+  const handleSave = async (savedSearchValues: Partial<SavedSearch>) => {
+    const body = {
+      body: {
+        ...savedSearchValues,
+        searchTerm,
+        filters,
+        count: totalResults,
+        searchPath: window.location.search,
+        sortField,
+        sortDirection
+      }
+    };
+
+    try {
+      if (search) {
+        await apiPut('/saved-searches/' + search.id, body);
+      } else {
+        await apiPost('/saved-searches/', body);
+      }
+      history.push('/inventory');
```

Comment:
I imagine you're reloading the page to ensure we don't have stale saved-searchs in the UI, could potentially change this behavior.
I see that we're using a context for the saved-search data, I think you could pass the fetchSearches into the context so that we could refetch them programmatically instead of reloading the view

Reply:
I agree. I was keeping the functionality as it was, but refactoring the way this refreshes is a good idea. It's been resolved